### PR TITLE
MM-36655,MM-36656,MM-36657: CRT beta texts and modal

### PR DESCRIPTION
**Pull request by koox00**
#### Summary

Shows an early beta warning modal when users enable CRT in account
settings, the modal is shown upon closing the user settings modal.

Changes account settings CRT related texts.

Removes `default_on` option for CRT in system console.

Adds `variant` prop to `AlertBanner` component so we can use it in the app as well as the system console

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-36655
https://mattermost.atlassian.net/browse/MM-36656
https://mattermost.atlassian.net/browse/MM-36657

#### Screenshots

image
image

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/collapsed_reply_threads_modal/index.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export {default} from './collapsed_reply_threads_modal';
```
Comment (devinbinnie) [inline components/collapsed_reply_threads_modal/index.ts:4]: We don't actually need this file if we're not connecting to redux or passing in any props or actions using it.

**Review by devinbinnie — APPROVED** [review]
Just one non-blocking comment.

**Review by esethna — APPROVED** [review]
Awesome, thanks @koox00 

Adding @jwilander as well if he can give a quick review.

@amyblais please make sure this gets in for cloud June 30 and on prem v5.37 (low risk changes that add a confirmation modal when the feature is enabled and updates some text)

**Comment by esethna** [comment]
@jgilliam17 please help give a priority review here

**Review by jwilander — APPROVED** [review]
Great work turning this around so quickly @koox00 !

**Review by jgilliam17 — APPROVED** [review]
Thanks @koox00 
Tested, looks good to merge.
- Verified CRT settings “default_off” and “off” are the only configurations available in System Console;  description text updated accordingly (MM-36657) 
- Verified Account Setting for CRT description text has been updated. (MM-36656)
- Verified each time user turns CRT setting On in the Account settings and closes the settings modal, early beta CRT warning is present (MM-36655)
- Small UX improvement/suggestion - “Got it” button is missing hover state, not sure if this is as designed.  @esethna Let me know and I can create a follow up for this if needed.
 

https://user-images.githubusercontent.com/52937121/123680283-82c8a600-d816-11eb-8c69-85a13a95e617.mov

**Pull request merged**